src/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -0,0 +1,675 @@
+        public int LastIndexOf(JsonNode item) => _list.LastIndexOf(item);
+
+        /// <summary>
+        ///   Removes all elements from the JSON array.
+        /// </summary>
+        public void Clear() => _list.Clear();
+
+        /// <summary>
+        ///   Removes the first occurrence of a specific object from the JSON array.
+        /// </summary>
+        /// <param name="item">
+        ///   The object to remove from the JSON array. The value can be null and it represents null JSON node.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if the item is successfully found in a JSON array and removed,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool Remove(JsonNode item) => _list.Remove(item);
+
+        /// <summary>
+        ///   Removes all the elements that match the conditions defined by the specified predicate.
+        /// </summary>
+        /// <param name="match">
+        ///   Thepredicate delegate that defines the conditions of the elements to remove.
+        /// </param>
+        /// <returns>The number of elements removed from the JSOJN array.</returns>
+        public int RemoveAll(Predicate<JsonNode> match) => _list.RemoveAll(match);
+
+        /// <summary>
+        ///   Removes the item at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">
+        ///   The zero-based index of the element to remove.
+        /// </param>
+        public void RemoveAt(int index) => _list.RemoveAt(index);
+
+        /// <summary>
+        ///   Copies the JSON array or a portion of it to an array.
+        /// </summary>
+        /// <param name="array">
+        ///   The one-dimensional Array that is the destination of the elements copied from JSON array.
+        ///   The Array must have zero-based indexing.
+        /// </param>
+        /// <param name="arrayIndex">The zero-based index in array at which copying begins.</param>
+        void ICollection<JsonNode>.CopyTo(JsonNode[] array, int arrayIndex) => _list.CopyTo(array, arrayIndex);
+
+        /// <summary>
+        ///   Returns an enumerator that iterates through the JSON array values.
+        /// </summary>
+        /// <returns>An enumerator structure for the <see cref="JsonArray"/>.</returns>
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        /// <summary>
+        ///   Returns an enumerator that iterates through the JSON array values.
+        /// </summary>
+        /// <returns>An enumerator structure for the <see cref="JsonArray"/>.</returns>
+        public IEnumerator<JsonNode> GetEnumerator() => new JsonArrayEnumerator(this);
+
+        /// <summary>
+        ///   Creates a new JSON array that is a copy of the current instance.
+        /// </summary>
+        /// <returns>A new JSON array that is a copy of this instance.</returns>
+        public override JsonNode Clone() => new JsonArray(_list);
```

Comment:
So a shallow clone?  Is that what we want?

Reply:
This should call `JsonArray(IEnumerable<JsonNode> values)`, which should call `new List<JsonNode>(values)`, which according to documentation should "initialize a new instance of the List<T> class that contains elements copied from the specified collection".

Reply:
```C#
JsonArray inner = new JsonArray { 1, 2, 3 };
JsonArray outer = new JsonArray { (JsonNode)inner) };
JsonArray outerClone = (JsonArray)outer.Clone();
((JsonArray)outerClone[0]).Add(4);

int innerLength = inner.Length;
```

This code will produce 4.  A deep clone would produce 3.  The difference is that while the list in `outer` got copied (so adding to outer doesn't affect outerClone) it didn't copy the things inside the list, so the mutable `inner` value is now in both of them.

Reply:
I see, thanks!